### PR TITLE
[utils] post delayed tasks with TaskRunner

### PR DESCRIPTION
**Pull request by wgtdkp**
This commit adds the new `Post` API to the `TaskRunner` to execute a task in future.
We also replace the handy `Timers` in openthread controller with the `TaskRunner` API.

### CODE REVIEW

File: src/common/task_runner.cpp [hunk 1]
```diff
@@ -69,39 +70,62 @@ TaskRunner::~TaskRunner(void)
...
+void TaskRunner::Process(const MainloopContext &aMainloop)
+{
+    OTBR_UNUSED_VARIABLE(aMainloop);
+
+    uint8_t n;
+
+    // Read any data in the pipe.
+    while (read(mEventFd[kRead], &n, sizeof(n)) == sizeof(n))
+    {
+    }
```
Comment (gjc13) [inline src/common/task_runner.cpp:117]: Print an error here if the read returns an unexpected value.

**Review by gjc13 — APPROVED** [review]
Posting an arbitary task is really a desired API. LGTM.

File: src/common/task_runner.cpp [hunk 2]
```diff
@@ -69,39 +70,62 @@ TaskRunner::~TaskRunner(void)
...
-void TaskRunner::Process(const MainloopContext &aMainloop)
-{
-    if (FD_ISSET(mEventFd[kRead], &aMainloop.mReadFdSet))
+    if (!mTaskQueue.empty())
     {
-        uint8_t n;
+        auto  now     = std::chrono::steady_clock::now();
+        auto &task    = mTaskQueue.top();
+        auto  delay   = std::chrono::duration_cast<std::chrono::microseconds>(task.GetTimeExecute() - now);
+        auto  timeout = std::chrono::microseconds{aMainloop.mTimeout.tv_usec + 1000000 * aMainloop.mTimeout.tv_sec};
```
[the remark this answers is not in the document]
Comment (wgtdkp) [reply src/common/task_runner.cpp:93]: Seems not available in OTBR. I tend to make a method pair that cast between `struct timeval` and `milliseconds`/`microseconds` in `time.hpp`.

**Review by Moandor-y — APPROVED** [review]
LGTM :+1:

**Review by bukepo — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**